lib/providers/plate_weights.dart

```diff
@@ -0,0 +1,70 @@
+import 'package:flutter/widgets.dart';
+import 'package:wger/helpers/gym_mode.dart';
+
+class PlateWeights extends ChangeNotifier{
+  String unitOfPlate = 'kg';
+  bool flag = false;
+  num totalWeight = 0;
+  num barWeight = 20;
+  num convertTolbs = 2.205;
+  num totalWeightInKg = 0;
+  num barWeightInKg = 20;
+  List<num> selectedWeights = [];
+  List<num> kgWeights = [1.25, 2.5, 5, 10, 15, 20, 25];
+  List<num> lbsWeights = [2.5, 5, 10, 25, 35, 45];
+  List<num> customPlates = [];
+  late Map<num,int> grouped;
+
+  void unitChange() {
+    if(unitOfPlate=='lbs') {
+      totalWeight = totalWeightInKg;
+      unitOfPlate = 'kg';
+      barWeight = barWeightInKg;
+    } else {
+      unitOfPlate = 'lbs';
+      totalWeight = totalWeightInKg*2.205;
+      barWeight = barWeightInKg*2.205;
+    }
+    notifyListeners();
+  }
+
+  void toggleSelection(num x) {
+    if(unitOfPlate == 'kg') {
+      if(selectedWeights.contains(x)) {
+        selectedWeights.remove(x);
+      } else {
+        selectedWeights.add(x);
+      }
+    } else {
```

Comment:
aren't the if and else branches the same? :D

In this respect, I think the different variables could be simplified, e.g.having always totalWeight and totalWeightKg seems redundant.